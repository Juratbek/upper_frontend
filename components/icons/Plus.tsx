import { IIconProps } from './types';

export const PlusIcon = (props: IIconProps) => (
  <div role='icon' {...props}>
    +
  </div>
);
