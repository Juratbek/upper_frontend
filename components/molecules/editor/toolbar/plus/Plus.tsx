import { FC } from 'react';
import { ICONS } from 'variables/icons';

import classes from './Plus.module.scss';

const PlusIcon = ICONS.plus;

export const Plus: FC<{ onClick: VoidFunction }> = ({ onClick }) => (
  <button className={classes.btn} onClick={onClick}>
    <PlusIcon width={32} height={32} strokeWidth={3} />
  </button>
);
