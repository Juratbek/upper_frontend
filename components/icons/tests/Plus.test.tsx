import { render, screen } from '@testing-library/react';

import { PlusIcon } from '../Plus';

describe('Plus icon', () => {
  it('renders the icon', () => {
    render(<PlusIcon />);
    const icon = screen.getByRole('icon');
    expect(icon).toMatchSnapshot();
  });
});
