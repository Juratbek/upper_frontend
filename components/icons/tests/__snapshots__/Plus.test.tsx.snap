// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Plus icon > renders the icon 1`] = `
<div
  role="icon"
>
  +
</div>
`;
